### PR TITLE
Migrate /profile/me endpoint and enable `strictNullChecks`

### PR DESCRIPTION
**Pull request by apalchys**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [ ] I followed naming convention rules

---

#### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

_Describe the source of requirement, like related issue link._

#### 💡 Background and solution

_Describe the big picture of your changes here_

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Database migration is added or not needed
- [ ] Documentation is updated/provided or not needed
- [ ] Changes are tested locally

### CODE REVIEW

File: nestjs/src/notifications/notifications.service.ts [hunk 1]
```diff
@@ -91,7 +93,7 @@ export class NotificationsService {
     });
   }
 
-  public buildChannelMessage(channel: NotificationChannelSettings & { externalId: string }, data: object) {
+  public buildChannelMessage(channel: NotificationChannelSettings & { externalId?: string }, data: object) {
```
Comment (aaliakseyenka) [inline nestjs/src/notifications/notifications.service.ts:96]: curious, what is the use case? seems this should be catched on upper calls

File: nestjs/src/profile/dto/update-profile.dto.ts [hunk 2]
```diff
@@ -142,50 +142,50 @@ class GeneralInfo {
...
   telegram: string | null;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
   @IsOptional()
   @IsString()
   notes: string | null;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
```
Comment (aaliakseyenka) [inline nestjs/src/profile/dto/update-profile.dto.ts:188]: just curious, won't it all be strings by definition?

File: nestjs/tsconfig.json [hunk 3]
```diff
@@ -13,7 +13,8 @@
     "paths": { "@entities/*": ["../server/src/models/*"] },
     "incremental": true,
     "skipLibCheck": true,
-    "strictNullChecks": false,
+    "strictNullChecks": true,
```
Comment (aaliakseyenka) [inline nestjs/tsconfig.json:16]: like

**Review by aaliakseyenka — APPROVED** [review]

File: nestjs/src/profile/dto/update-profile.dto.ts (hunk 2 above)
Comment (apalchys) [reply nestjs/src/profile/dto/update-profile.dto.ts:188]: it fails to detect type if you use union `string | null`. It becomes `object` :(

File: nestjs/src/notifications/notifications.service.ts (hunk 1 above)
Comment (apalchys) [reply nestjs/src/notifications/notifications.service.ts:96]: yes, not needed after rebase. removed

**Comment by apalchys** [comment]
@aaliakseyenka I also did some renames to follow nestjs `kebab-case` notation for files/folders

**Pull request merged**